### PR TITLE
[ROCm][CI] Fix test_token_classification.py::test_bert_models

### PR DESCRIPTION
**Pull request by divakar-amd**
This PR fixes **mi325_1: Language Models Test (Extended Pooling):** `models/language/pooling/test_token_classification.py::test_bert_models[float-boltuix/NeuroBERT-NER]`

The issues were caused due to similar reasons mentioned in https://github.com/vllm-project/vllm/pull/31612

---

> [!NOTE]
> Addresses ROCm-specific correctness in token classification tests.
> 
> - For `test_bert_models` and `test_modernbert_models`, pass `model_kwargs={'attn_implementation': 'eager'}` to HF `AutoModelForTokenClassification` when `current_platform.is_rocm()` to avoid flash attention accuracy issues
> - In `test_bert_models`, switch from `torch.allclose(..., 1e-2)` to `torch.testing.assert_close(..., atol=1.2e-2, rtol=1e-3)` for logits comparison
> - Minor refactor: import `current_platform` at top-level (remove local import)
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit 7744528ffb4a97421d26753c7bafa51fae7eadbd. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>

### CODE REVIEW

**Review by noooop — APPROVED** [review]

**Comment by noooop** [comment]
Please try merging the branch 'main' into this PR to resolve the buildkite/amd-ci failure.

**Review by DarkLight1337 — APPROVED** [review]
